### ISSUE TITLE
Crate needs rustdoc

### ISSUE THREAD

**Issue by BartMassey** [dylagit/audio-limiter#1]
Would really be nice to have some things commented in the crate, to make it easier to figure out what is going on and add stuff. In particular, it would be nice to document the source of the compressor algorithm used in `compressor.rs`.

**Comment by dylagit** [comment]
Yeah, this is a good idea. I mostly stitched together `compressor.rs` through advice in Discord and piecing together snippets found online and on GitHub, so unfortunately I don't really understand the algorithm either! However, I do understand the rest of the codebase so I'll look into adding some comments in the other files.

**Comment by BartMassey** [comment]
Please do at least document the sources you found online and on GitHub as best you can.

Thanks much for this crate!
